code/game/gamemodes/game_mode.dm

```diff
@@ -445,4 +445,14 @@ proc/get_nt_opposed()
 	for(var/obj/machinery/nuclearbomb/bomb in world)
 		if(bomb && bomb.r_code && bomb.z == ZLEVEL_STATION)
 			nukecode = bomb.r_code
-	return nukecode
\ No newline at end of file
+	return nukecode
+
+/datum/game_mode/proc/replace_jobbaned_player(mob/living/M, role_type, pref)
+	var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as a [role_type]?", "[role_type]", null, pref, 100)
+	var/mob/dead/observer/theghost = null
+	if(candidates.len)
+		theghost = pick(candidates)
+		M << "Your mob has been taken over by a ghost! Appeal your job ban if you want to avoid this in the future!"

```

Comment:
"Stop being a little shit, you scumbag, and appeal your job-ban, if you want to avoid this in the future"

Totally should be what it says =p